### PR TITLE
3900 action hubspot enroll in a workflow or a sequence

### PR DESCRIPTION
**Pull request by luancazarine**
Action:
  - Enroll Contact Into Workflow

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/3984"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/hubspot/common/constants.mjs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import {
-  OBJECT_TYPE,
+  HUBSPOT_OWNER, OBJECT_TYPE,
```
Comment (andrewjschuang) [inline components/hubspot/common/constants.mjs:2]:
```suggestion
  HUBSPOT_OWNER,
  OBJECT_TYPE,
```

File: components/hubspot/hubspot.app.mjs [hunk 2]
```diff
@@ -1,11 +1,8 @@
 import { axios } from "@pipedream/platform";
 import {
-  ASSOCIATION_CATEGORY,
-  HUBSPOT_OWNER,
+  API_PATH, ASSOCIATION_CATEGORY, BASE_URL, HUBSPOT_OWNER,
```
Comment (andrewjschuang) [inline components/hubspot/hubspot.app.mjs:3]:
```suggestion
  API_PATH,
  ASSOCIATION_CATEGORY,
  BASE_URL,
  HUBSPOT_OWNER,
```

File: components/hubspot/hubspot.app.mjs [hunk 3]
```diff
@@ -667,5 +680,24 @@ export default {
...
     },
+    async listWorkflows($) {
+      return this.makeRequest(
+        API_PATH.AUTOMATION,
+        "/workflows",
+        {
+          $,
+        },
+      );
+    },
```
Comment (andrewjschuang) [inline components/hubspot/hubspot.app.mjs:691]: If the only place this method is called is from `async options`, then there's no need for the `$` param.

```suggestion
    async listWorkflows() {
      return this.makeRequest(
        API_PATH.AUTOMATION,
        "/workflows",
      );
    },
```

File: components/hubspot/hubspot.app.mjs [hunk 4]
```diff
@@ -185,6 +182,22 @@ export default {
         };
       },
     },
+    workflow: {
+      type: "string",
+      label: "Workflow",
+      description: "The ID of the workflow you wish to see metadata for.",
+      async options() {
```
Comment (andrewjschuang) [inline components/hubspot/hubspot.app.mjs:189]: Should this support pagination with `context`?

File: components/hubspot/hubspot.app.mjs [hunk 5]
```diff
@@ -185,6 +182,22 @@ export default {
...
+        const { workflows } = await this.listWorkflows();
+        return {
+          options: workflows.map(({
+            name: label, id: value,
+          }) => ({
+            label,
+            value,
+          })),
+        } || [];
```
Comment (andrewjschuang) [inline components/hubspot/hubspot.app.mjs:198]: Is `|| []` necessary here?
Or should there be a check before return options?

**Review by andrewjschuang — COMMENTED** [review]
@luancazarine thanks for the contribution!

I've left just a few comments below.

File: components/hubspot/hubspot.app.mjs (hunk 4 above)
Comment (luancazarine) [reply components/hubspot/hubspot.app.mjs:189]: this API doesn't support pagination

**Review by andrewjschuang — APPROVED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[Hubspot_3900.pdf](https://github.com/PipedreamHQ/pipedream/files/9316886/Hubspot_3900.pdf)

**Pull request merged**